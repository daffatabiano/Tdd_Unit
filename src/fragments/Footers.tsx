export default function Footers() {
    const date = new Date();
    const year = date.getFullYear();
    return (
        <footer className="footer bottom-0 p-10 bg-secondary text-neutral-content">
            <div className="items-center grid-flow-col">
                <p>Copyright Â© {year} - All right reserved</p>
            </div>
        </footer>
    );
}
