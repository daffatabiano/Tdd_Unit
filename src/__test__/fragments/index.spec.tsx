import React from 'react';
import { render } from '@testing-library/react';
import Navbar from '@/fragments/Navbar';
import Footers from '@/fragments/Footers';

describe('Navbar components', () => {
    it('should navbar render', () => {
        const { container } = render(<Navbar />);
        expect(container).toMatchSnapshot();
    });
    // it('should navbar text', () => {
    //     const { getByText } = render(<Navbar />);
    //     expect(getByText('Navbar')).toBeInTheDocument();
    // });
});

describe('Footers components', () => {
    it('should footer render', () => {
        const { container } = render(<Footers />);
        expect(container).toMatchSnapshot();
    });
    // it('should footer text', () => {
    //     const { getByText } = render(<Footers />);
    //     expect(
    //         getByText('Copyright Â© 2024 - All right reserved')
    //     ).toBeInTheDocument();
    // });
});
