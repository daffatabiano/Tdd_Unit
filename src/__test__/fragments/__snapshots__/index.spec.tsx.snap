// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footers components should footer render 1`] = `
<div>
  <footer
    class="footer bottom-0 p-10 bg-secondary text-neutral-content"
  >
    <div
      class="items-center grid-flow-col"
    >
      <p>
        Copyright Â© 
        2024
         - All right reserved
      </p>
    </div>
  </footer>
</div>
`;

exports[`Navbar components should navbar render 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg bg-body-tertiary"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        <img
          alt=""
          height="30"
          loading="lazy"
          src="/logo-travel.png"
          width="30"
        />
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <form
          class="d-flex"
          role="search"
        >
          <input
            aria-label="Search"
            class="form-control me-2"
            placeholder="Search"
            type="search"
          />
          <button
            class="btn btn-outline-success"
            type="submit"
          >
            Search
          </button>
        </form>
      </div>
    </div>
  </nav>
</div>
`;
